config/crd/bases/config.nri_pluginconfigs.yaml

```diff
@@ -0,0 +1,68 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: pluginconfigs.config.nri
+spec:
+  group: config.nri
+  names:
+    kind: PluginConfig
+    listKind: PluginConfigList
+    plural: pluginconfigs
+    singular: pluginconfig
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: PluginConfig is the Schema for the pluginconfigs API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: Spec defines the intended state of PluginConfig
+            properties:
+              pluginName:
```

Comment:
Is the master plan to create multiple PluginConfigs in order to get multiple NRI plugins installed on nodes, or is the plan to allow specifying multiple plugins, versions and their target nodes in the same PluginConfig? Obviously it's a perfectly valid use case to run `balloons` on some nodes, `topology-aware` on all the other nodes, and `memory-qos` on all nodes, for instance. Let's make sure that the design will enable getting there at some point.

Reply:
I designed it like one `PluginConfig` is for installing one NRI plugin on a given namespace. We could extend the CRD to have a list of plugins but then it means they will get deployed on the same namespace and I think that's not a good situation to be in?